scripts/host-check

```diff
@@ -1749,16 +1793,37 @@ def run_checks_all_platforms(extra_checks: List[str]) -> bool:
         )
     if platforms_errored:
         print(
-            "!! One or more platform checks could not be performed, see errors above !!"
+            "!! One or more platform checks could not be performed for XR platforms:   !!"
+            + "\n      "
+            + ",\n      ".join(sorted(platforms_errored))
+            + "\n   See errors above."
         )
+    if platforms_warned:
+        print(
+            "!! One or more platform checks resulted in a warning for XR platforms:    !!"
+            + "\n      "
+            + ",\n      ".join(sorted(platforms_warned))
+            + "\n   See warnings above."
```

Comment:
I'm not a huge fan of these messages, but I think the problem I have is not with the messages themselves per-se, but the overall UX of the script when a specific platform is not given.

I don't think it makes sense for the user to ask "are any XRd platforms supported on this host?".  Generally I'd expect the user to know what platform they want to use, and ask "does this host support this platform".

I can potentially see the use-case for "does this host support both XRd platforms" - i.e., are all XRd platforms supported on this host?

What we've done as part of this PR is move closer to "all platforms" - if at least one platform is not supported, then we exit nonzero.

However the error messages still lean towards "any platform".  If I ask "are all XRd platforms supported on this host?", and CP is, but VR isn't, then I want the answer to be clearly "no".  Whereas now our answer (according to the messages emitted) is "XRd CP is supported.  XRd VR is NOT supported."  This is less clear and open to misinterpretation - e.g., you could easily just skim over the "supported" line and expect that both platforms are supported, or you could get CP and VR mixed up, etc.

I think the entire thing would be clearer if either:
* The user was forced to specify a platform
* If that's deemed to awkward a UX, and if the "all platforms" is a justified use case, then if at least one platform is not supported then make this a very clear "NO" summary, followed by details [if details are even necessary vs just referring to the individual lines of output].

I'm **not** suggesting to do this as part of this PR.  However I think it would be worth chatting with Paulo and seeing if a followup task is justified here to nail down some aspects of the UX/design of this script (AIUI this has kinda morphed organically from just CP, to CP + VR, to CP + VR + extra checks)

Reply:
Tagging @paulo-cisco for awareness.